fhir-bucket/src/main/java/com/ibm/fhir/bucket/scanner/CosReader.java

```diff
@@ -203,20 +215,21 @@ public void mainAllocationLoop() {
                 // We have more capacity than work is currently available in the database,
                 // so take a nap before checking again
                 logger.fine("No work. Napping");
-                safeSleep(5000);
+                safeSleep(10000L);
             }
         }
     }
     
     /**
-     * Sleep this thread for the given milliseconds
+     * Sleep current thread for given number of milliseconds or until
+     * the thread is interrupted.
      * @param millis
      */
     protected void safeSleep(long millis) {
         try {
             Thread.sleep(millis);
         } catch (InterruptedException x) {
-            // NOP
+            // woken up early from sleep, probably shutting down, so this is a NOP
         }
     }
```

Comment:
This is all over the place (e.g. repeated code)

Reply:
guilty. Will create a utility, not now.

Reply:
Something for the future.  agreed, marking resolved.